### ISSUE TITLE
3.1.2 Lang of Parts exceptions need to be Key Terms

### ISSUE THREAD

**Issue by jamieherrera** [w3c/wcag#4094]
It would be very helpful in practice to get Key Terms for the "except for" part of 3.1.2 Language of Parts. 

Second, I'd like to understand this team's intent for the phrasing of the 3.1.2 Language of Parts SC, primarily the reason to use "except for":

> The [human language](https://www.w3.org/WAI/WCAG22/Understanding/language-of-parts.html#dfn-human-language) of each passage or phrase in the content can be [programmatically determined](https://www.w3.org/WAI/WCAG22/Understanding/language-of-parts.html#dfn-programmatically-determined) except for proper names, technical terms, words of indeterminate language, and words or phrases that have become part of the vernacular of the immediately surrounding text.

This SC seems to imply the second clause is clearly not able to be programmatically determinable. Is the intent that the "except for" items not be pronounced by AT in their original language even if it can be pronounced?

- "can be" implies there are situations that "can't be" ; I know the 3.1.1 SC uses the same phrasing so maybe just ignore this bullet. 

- The "except for" part of the SC some people have interpreted as meaning that any of these exceptions are **exempt** from needing (or even, "should **never** use") a language attribute.

In the case I'm thinking of, a company uses several branding product names that are purposely not translated into a second language when the surrounding text is in that second language. It is a decision; these words are trademarked. That product name is clearly in a programmatically determinable language (say, clearly in English, which has a programmatically determinable language "en"). As the name of a "thing" it could be considered a "proper name" if using a generic definition from an internet search, like a "person, place, or thing". WCAG does not define "proper name". It seems to do AT users, particularly SR users, a disservice by resulting in poor pronunciation for items that are clearly one language's version of a noun.  If it used a lang attribute, AT users would understand this product is the same product as what is advertised on commercials; if not using a lang attribute some SR butcher the words as unintelligible in the second language.

Also, most languages have alternative proper nouns for places and things, like "Twin Towers" in English is "Torres Gemelas" in Spanish; it was a place, it has proper nouns, but the proper nouns can change. 

TL; DR
I would strongly suggest formally adding each exception as its own Key Term. 
I would strongly suggest limiting the "proper name" term to specific use cases
I would strongly suggest a sentence in the Understanding document that permits using a programmatically determinable language on these exceptions (as defined in a key term) if the author determines it may serve the intended audience, but that it would not fail the SC.

**Comment by patrickhlauke** [comment]
> The "except for" part of the SC some people have interpreted as meaning that any of these exceptions are exempt from needing (or even, "should never use") a language attribute

> I would strongly suggest a sentence in the Understanding document that permits using a programmatically determinable language on these exceptions (as defined in a key term) if the author determines it may serve the intended audience, but that it would not fail the SC.

this seems to misunderstand the purpose of exemptions. the SCs set a bottom line, a floor, for determining if something passes or fails WCAG. the exemptions say that in those situations, if something isn't done, it doesn't *fail* an SC. it does *not* mean the opposite ... that if authors *do* do it, they *fail*. It just means that it's not necessary to do it in those cases to *pass*.

**Comment by jamieherrera** [comment]
What is the definition of a "proper name" @patrickhlauke for WCAG 3.1.1 and 3.1.2? Is it a narrower interpretation than "proper noun", such as a person's name?
If I have a page in Spanish and purposely leave the words, "United States" instead of "estados unidos", would that not require a lang attribute? It's clearly English on a Spanish page, and it's clearly a proper noun as well.

**Comment by jamieherrera** [comment]
The term "United States"  can be programmatically determined as well.

**Comment by patrickhlauke** [comment]
the difference/overlap between proper name and proper noun is a long-standing point of discussion (I remember last time I looked at this, stumbled across https://en.wikipedia.org/wiki/Wikipedia:Proper_names_and_proper_nouns as one example of lots of debate). leaving it up to the rest of the working group to weigh in here...

(but on the "United States" can be programmatically determined comment: it's not really programmatically determinable if there is no machine-readable/understandable extra information there, which is what programmatically determined means. sure, I could run every single word against dictionaries for every possible language, and then infer what language it's in based on the most likely matches of this search ... but that brute-force/heuristic approach is not what "programmatically determined" means)

**Comment by jamieherrera** [comment]
Programmatically determinable here doesn’t just mean that lang = “en”?

The main issue I have is that without a definition for proper name, this
exception is being extended to terms that are clearly a proper noun (not
someone’s name) and unclear why it merits an exception. If a proper noun
has multiple distinct language variations (like the proper noun “United
States”, in English, lang=“en”; “estados unidos” clearly Spanish needs
lang= “es”) if someone insists on not translating to the page language
citing it’s a “proper name” they are missing the point of the SC.
My primary ask is to add a key term for “proper name” to the WCAG key
terms, and secondarily flesh this out in the Understanding document as has
already been done with “technical term” and “words or phrases that have
become part of the vernacular of the immediately surrounding text.”
This is about the ability for content to be communicated as intended for
people using assistive technology.
